### PR TITLE
Documentation sprint - Metadata.md

### PR DESCRIPTION
**Pull request by rosiel**
**GitHub Issue**: https://github.com/Islandora/documentation/issues/1040

# What does this Pull Request do?

Updates documentation for Metadata!

# What's new?

Not sure yet, this is a WIP.

# How should this be tested?

Read it, and leave comments if it's wrong or confusing!

# Interested parties
[I'll tag collab, Seth and Wilhelmina, when it's ready, but you're free to jump in now if you want!)

### CODE REVIEW

File: docs/user-documentation/metadata.md [hunk 1]
```diff
@@ -1,15 +1,17 @@
 # Metadata in Islandora 8
 
-> TL;DR: In Islandora 8 metadata values are stored in _fields_ attached to _entities_ (objects) which are then serialized as JSON-LD before being submitted to Fedora and/or indexed in a triple-store.
+> TL;DR: In Islandora 8, metadata is stored in _fields_ attached to _entities_ (nodes or media). Metadata values are serialized as RDF in the form of JSON-LD before being submitted to Fedora and/or indexed in a triplestore.
```
Comment (mjordan) [inline docs/user-documentation/metadata.md]: I know that serialization into RDF is an important feature of Islandora 8, but I also feel that we as a community have been giving it a more dominant place in our discussions around metadata than I think it deserves. I say this because by making RDF serialization into JSON-LD that is then indexed in a triplestore so dominant a characteristic of the way that Islandora handles metadata, we're removing the focus from the simpler, more concrete idea that metadata in Islandora 8 is just plain old Drupal fields, and one very useful, but somewhat esoteric (let's be honest), feature of Islandora 8 is that the data in the fields can be expressed as RDF-in-JSON-LD-in-a-triplestore.

Comment (seth-shaw-unlv) [reply docs/user-documentation/metadata.md]: I do think it deserves the more dominant place... as a repository much of our focus is getting our objects into a long-term persistence store (Fedora) and, currently, only the metadata that is serialized as JSON-LD gets that privilege.

I agree that the triple-store is of secondary importance (I only view it as an access point), but it is the JSON-LD to Fedora chain that earns it this place of importance.

File: docs/user-documentation/metadata.md [hunk 2]
```diff
@@ -1,15 +1,17 @@
...
 
 !!! note "Drupal 8 Terminology"
-    In Drupal 8, Fields can be attached to _entity sub-types_ (e.g. Content types, Vocabularies) or _entities_ (Users, Files). For more on Fields, see ["2.3 Content Entities and Fields"](https://www.drupal.org/docs/user_guide/en/planning-data-types.html) and ["6.3 Adding Basic Fields to a Content Type"](https://www.drupal.org/docs/user_guide/en/structure-fields.html) in the Official Guide.
+    In Drupal 8, Fields can be attached to _entity sub-types_ [(formerly called _bundles_](https://www.drupal.org/project/drupal/issues/1380720) (e.g. Content types, Media types, Vocabularies) or _entities_ (Users, Files). For more on Fields, see ["2.3 Content Entities and Fields"](https://www.drupal.org/docs/user_guide/en/planning-data-types.html) and ["6.3 Adding Basic Fields to a Content Type"](https://www.drupal.org/docs/user_guide/en/structure-fields.html) in the Official Drupal Guide.
```
Comment (dannylamb) [inline docs/user-documentation/metadata.md]: Not your words (probably mine), but Files are not fieldable entities.

Comment (seth-shaw-unlv) [inline docs/user-documentation/metadata.md]: Hmmm, 'bundles' is still the official terminology. The issue listed is to _perform usability testing_ on the change rather than implement it (and the ticket is still unassigned). Perhaps this ticket can be put in a comment so we can revisit it later when we revise documentation again:
```
<!-- Next revision: check status of changing 'bundles' to 'entity sub-types' (https://www.drupal.org/project/drupal/issues/1380720). -->
```
That way the docs will reflect practice as it currently stand and won't confuse current readers but we are still aware of the possible change when we look at the source markup.

File: docs/user-documentation/metadata.md [hunk 3]
```diff
@@ -1,15 +1,17 @@
...
+In Drupal, _Content Types_ are sub-types of _Nodes_. Content types contain fields and configurations for how those fields can be edited or displayed. Each content type is essentially a _metadata profile_ for a digital resource's descriptive record. For each field in a content type, an administrator can configure how data is entered, how it can be displayed, how many values can be stored, and how long the value can be. Some configurations, such as data entry and display, can be changed at any time. Others, such as how long a value can be or what options are available in a Select list, cannot be changed once content has been created without first deleting all content of that type.
+
+<!-- I am here -->
```
Comment (seth-shaw-unlv) [inline docs/user-documentation/metadata.md]: Should probably not be here. 😉

File: docs/user-documentation/metadata.md (hunk 1 above)
Comment (mjordan) [reply docs/user-documentation/metadata.md]: Good points, but I guess it depends on who the audience is. Does the average repo manager need to know about the plumbing that gives JSON-LD this privileged role and importance in Islandora, as long as everything just works? I'm asking that in as neutral a way as possible.

Comment (seth-shaw-unlv) [reply docs/user-documentation/metadata.md]: I do think it is important to recognize this at a high-level that there is a difference between what is in your Drupal and what shows up in your Fedora. They don't necessarily need to know the plumbing (RDF-Mapping → JSON-LD Serializer → Normalize JSON-LD Array Alters) but the librarians here, at least, appreciate knowing there is a jump from Drupal to JSON-LD to Fedora.

File: docs/user-documentation/metadata.md (hunk 2 above)
Comment (seth-shaw-unlv) [reply docs/user-documentation/metadata.md]: It might have been my words as well; I know I touched this document quite a bit the last go-around. I think the confusion may have come from the fact that File does extend ContentEntityBase and, thus, has _fields_ but is **not** _fieldable_ (as you noted).

File: docs/user-documentation/metadata.md (hunk 1 above)
Comment (mjordan) [reply docs/user-documentation/metadata.md]: I'm not saying we shouldn't document that all that exists, I'm saying I think we've given those aspects of Islandora undue emphasis compared to the fact that the canonical version of the metadata, the one that most users interact with directly, is just plain old Drupal fields. It's just a matter of emphasis.

Comment (seth-shaw-unlv) [reply docs/user-documentation/metadata.md]: Going back to the 'depends on who the audience is'; we have two sides of a coin here for perspective, the day-to-day 'access' side which is more interested in the Drupal fields with their editing and presentation and the 'preservation' side which is very concerned about what metadata is serialized. I think the overall documentation gives both concepts a good amount of coverage.

I think this particular line of documentation that spawned this discussion should also represent both sides of this coin. So, if we are to edit it, I suggest we _add_ a bit more emphasis to the 'Drupal fields is what we display to users' aspect, rather than take away from the 'Fields get serialized to JSON-LD for Fedora' aspect.

_Etymological side note:_

I don't know that I would call the Drupal fields 'canonical'. Certainly, the Drupal fields are what most users interact with directly, as you say, but I think the metadata as it is serialized to Fedora better fits the classical definition of 'canonical' where its role in preservation is a reflection of the term's history of reflecting a 'sacred' nature.

Comment (rosiel) [reply docs/user-documentation/metadata.md]: The MIG has spent 2 years mapping to RDF, not to fields. I wanted the word "RDF" in here so there is a connection to all the predicates we've spent so long debating. 

Can I reword it to "Metadata values _can be_ serialized" rather than "_are_ serialized"?

File: docs/user-documentation/metadata.md (hunk 2 above)
Comment (rosiel) [reply docs/user-documentation/metadata.md]: haha thanks for catching that too. I had pinged Seth for confirmation as soon as I saw that. :P

Comment (rosiel) [reply docs/user-documentation/metadata.md]: So, at the docs _stood_ they said "entity sub_types". My first thought was "hey, that should be bundles!" but then I looked it up and yeah there's been a push since 2011 to change it to "entity sub-types". Thanks, i'll revise, and good suggestion on the comment.

File: docs/user-documentation/metadata.md (hunk 3 above)
Comment (rosiel) [reply docs/user-documentation/metadata.md]: Yo, it's a _draft_.

File: docs/user-documentation/metadata.md (hunk 1 above)
Comment (mjordan) [reply docs/user-documentation/metadata.md]: It's fine as is. I was simply making a point about emphasis.

Comment (rosiel) [reply docs/user-documentation/metadata.md]: It was a good point to make, thank you!

File: docs/user-documentation/metadata.md [hunk 4]
```diff
@@ -1,23 +1,28 @@
...
+    Note that the "Title" field does not appear in this list, as it is built-in and included with every content type. You can edit the label if you want it called something other than "Title" under the "Edit" tab for that content type. This built-in title "field" is limited to 255 characters; if your content has longer titles it is encouraged to create a separate long_title field to store the full title and reserve the default title field for a display title. There is also a contributed module called [Node Title Length](https://www.drupal.org/project/title_length) but we do not know if anyone in the community is using it.
 
+<!-- I am here. --> 
 !!! tip "7.x Migration Note: What about my XML?"
     In 7.x, metadata were (usually) stored within XML datastreams such as MODS or DC. In Islandora 8 we are breaking out
     individual metadata elements into fields instead of using an attached XML document. The Metadata Interest Group is developing a default mapping which will provide a basic, yet customizable, transform between MODS metadata and Drupal fields in Islandora Demo.
```
Comment (manez) [inline docs/user-documentation/metadata.md]: Islandora Demo -> islandora_defaults?

File: docs/user-documentation/metadata.md (hunk 2 above)
Comment (rosiel) [reply docs/user-documentation/metadata.md]: Marking resolved - current state is:
`Fields can be attached to _bundles_ (sometimes called _entity sub-types_ -- e.g. Content types, Media types, Vocabularies)`

**Comment by rosiel** [comment]
Hey folks thank you for your comments on the draft! The PR is marked as "ready for review" now. I know I missed the sprint deadline (sorry) but I don't want this to get orphaned.

File: docs/user-documentation/metadata.md [hunk 5]
```diff
@@ -1,83 +1,130 @@
...
+    This means we are breaking out individual elements from a hierarchical structure to being individual independent values. Where some hierarchy or field grouping
+    is necessary, this can be done in Drupal using [Paragraphs](https://www.drupal.org/project/paragraphs), a widely-used Drupal contrib module. At the moment
+    (Nov 2019) we are working on the technical challenge of mapping data from paragraphs into RDF in Fedora.
+    The Metadata Interest Group has developed a default mapping ([spreadsheet](https://docs.google.com/spreadsheets/d/18u2qFJ014IIxlVpM3JXfDEFccwBZcoFsjbBGpvL0jJI/edit#gid=0), [guidance document](https://docs.google.com/document/d/15qSO9YcALtYSqd6CUuGx0t8FwUJ5pPwVPz0PA5rU898/edit?ts=5c5852f3#)) which provides a basic, yet customizable, transform between MODS metadata and Drupal fields in Islandora Defaults. It is suggested that individual institutions customize the mapping to meet their unique needs.
+
+    That said, if keeping the "legacy" XML metadata from 7.x is important to you, it can be attached to an Islandora 8 resource node as a Media entity.
```
Comment (manez) [inline docs/user-documentation/metadata.md:32]: When I preview this page, this paragraph is broken out into a box with a horizontal scroll. I can't figure out where this is coming from based on the source text, but figured it was worth mentioning.

Comment (rosiel) [reply docs/user-documentation/metadata.md:32]: Yeah, that's really weird. It's only in Github, because the `!!! tip` code is mkdocs-specific. It renders correctly in Mkdocs. The existing docs do the same thing.  https://github.com/Islandora/documentation/blob/master/docs/user-documentation/metadata.md 
Screen Shot 2019-11-29 at 1 05 03 PM

**Review by manez — APPROVED** [review]
This should probably have an additional review by some one who understand the concepts better, but I found it very readable and informative. Good docs!

File: docs/user-documentation/metadata.md [hunk 6]
```diff
@@ -1,83 +1,130 @@
...
 
-For example, the 'islandora_demo' module provides a _Repository Item_ content type that defines several fields including "Alternative Title" and "Date". The "Manage fields" page for Repository Item shows a list of the fields it includes as well as tabs for changing the input forms and display modes. The "Title" field does not appear in this list but it is still included with every content type. (See the "[Create / Update a Content Type](content_types.md)" section for more details on creating and configuring fields.)
 
-![Screenshot of the "Manage fields" page for the "Repository Item" content type from islandora_demo.](../assets/metadata_content_type_screenshot.png)
+For example, the 'islandora_defaults' module provides a _Repository Item_ content type that defines many fields including "Alternative Title" and "Date Issued". Under the management menu for Repository Item you can see a list of the fields it includes ("Manage fields" as well as tabs for changing the input forms ("Manage form display") and display modes ("Manage display"). (See the "[Create / Update a Content Type](content_types.md)" section for more details on creating and configuring fields.)
```
Comment (dannylamb) [inline docs/user-documentation/metadata.md]: Missing a parenthesis at the end of "Manage fields".  Also, the last sentence doesn't have to be parenthetical.

File: docs/user-documentation/metadata.md [hunk 7]
```diff
@@ -1,83 +1,130 @@
...
 
-The above sections described how Drupal manages and stores metadata, but the key feature of Islandora 8 is pushing that metadata into a Fedora 4+ repository and a triple-store. Islandora does this by using Drupal's serialization capabilities to provide a JSON-LD serialization that can be ingested by Fedora 4+ repository and triple-stores. In response to write operations, it sends notifications to the repository and triple-store that a node or term is available to ingest.
+The above sections described how Drupal manages and stores metadata, but Islandora 8 provides for pushing that metadata into a Fedora 4+ repository and a triple-store. Islandora does this by using Drupal's serialization capabilities to provide a JSON-LD serialization that can be ingested by Fedora 4+ repository and triple-stores. In response to write operations, it sends notifications to the repository and triple-store that a node or term is available to ingest.
 
 The JSON-LD module works by taking node or term and its corresponding RDF mapping to create a JSON-LD serialization. The RDF mapping for a content type or vocabulary lists its fields and the RDF predicates that should be used for them.
```
Comment (dannylamb) [inline docs/user-documentation/metadata.md]: Not your words but, "taking node or term" is missing an "a".  e.g. "taking a node or term".  Or even better, just say "taking an entity".

**Review by dannylamb — CHANGES_REQUESTED** [review]
Looks accurate to me.  Found a few small things but nothing major.

File: docs/user-documentation/metadata.md (hunk 7 above)
Comment (rosiel) [reply docs/user-documentation/metadata.md]: I know, I rushed that last part. Thanks.

Comment (rosiel) [reply docs/user-documentation/metadata.md]: it now reads:

The JSON-LD module (written for Islandora) takes an entity and its corresponding RDF mapping (a structure defined by the contributed RDF module) to create a JSON-LD serialization. An RDF mapping for a content type, media type, or vocabulary lists its fields and the RDF predicates that should be used for them.

The JSON-LD serialization for an entity is available by appending `?_format=jsonld` to the entity's URL. This particular syntax takes advantage of the REST API module. Below is an example JSONLD document representing the RDF serialization of a Repository item node created in a standard islandora-playbook based vagrant VM:

**Review by dannylamb — APPROVED** [review]
:+1:
